src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -780,9 +821,154 @@ protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRe
                         final int clipFrom = neg.getReadLength() - negDiff - clipped + 1;
                         CigarUtil.softClip3PrimeEndOfRead(neg, Math.min(neg.getReadLength(), clipFrom));
                     }
+
+                }
+            }
+        }
+    }
+
+    protected static int getReadPositionToClipFrom(final SAMRecord rec, final int refPosToClipFrom) {
```

Comment:
hmmm. this feels like code that already exists in htsjdk....basically you want to go from a reference position to a readoffset, right? how about htsjdk.samtools.SAMRecord#getReadPositionAtReferencePosition(htsjdk.samtools.SAMRecord, int, boolean) does that help?

Reply:
comment still stands